src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -146,6 +151,7 @@ public void onConnectionResumed(boolean sessionPresent) {
 
     CompletableFuture<Integer> publish(MqttMessage message, QualityOfService qos, boolean retain) {
         return connect().thenCompose((b) -> {
+            rateLimiter.acquire(1);
             logger.atTrace().kv(TOPIC_KEY, message.getTopic()).kv(QOS_KEY, qos.name()).kv("retain", retain)
```

Comment:
this isn't part of your change - but should this log move into the synchronized block? if multiple threads are waiting on the sync block, these logs could be out of order